server/clearbit.js

```diff
@@ -95,10 +97,44 @@ export default class Clearbit {
 
     this.metric("saveUser");
 
-    return this.hull
-      .as(ident)
-      .traits(traits)
-      .then(() => { return { user, person }; });
+    if (this.settings.handle_accounts) {
+      const all_traits = { user: {}, account: {} };
+
+      _.map(traits, (v, k) => {
+        const [group, trait] = k.split("/");
+        if (group === "clearbit_company") {
+          all_traits.account[`clearbit/${trait}`] = v;
+        } else {
+          all_traits.user[k] = v;
+        }
+      }, {});
+
+      const domain = all_traits.account["clearbit/domain"];
+
+      // Set top level traits
+      const top_level_traits = {
+        "name": "name",
+        "domain": "domain",
+      };
+      _.forIn(top_level_traits, (clearbit_name, top_level_name) => {
+        const value = all_traits.account[`clearbit/${clearbit_name}`];
+        if (value) {
+          _.set(all_traits.account, top_level_name, { value, operation: "setIfNull" });
+        }
+      });
+
+      const client = this.hull.asUser(ident);
+
+      client.traits(all_traits.user);
```

Comment:
If you need to resolve to `{ user, person }` you can do
`return Promise.all(promises).then(() => { return { user, person }; })`